### PR TITLE
Implemented safe handling of messages and implemented moving messages.

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerMovedMessageHandler.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package nl.soccar.gameserver.message.handler;
+
+import io.netty.buffer.ByteBuf;
+import java.util.logging.Logger;
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerMovedMessageHandler.java:4]: MINOR Remove this unused import 'java.util.logging.Logger'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java [hunk 2]
```diff
@@ -144,26 +148,33 @@ public boolean leaveSession(Player player, TeamColour colour, Session session) {
...
+
         Team teamBlue = room.getTeamBlue();
-        
+
         TeamColour colour = teamBlue.getPlayers().stream().filter(p -> p.getUsername().equals(player.getUsername())).count() > 0 ? TeamColour.BLUE : TeamColour.RED;
         leaveSession(player, colour, session);
     }
-    
+
     public void startGame(Session session) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java:161]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java [hunk 3]
```diff
@@ -144,26 +148,33 @@ public boolean leaveSession(Player player, TeamColour colour, Session session) {
...
         room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> c.send(m));
-        
+
         Game game = session.getGame();
         game.start();
+
+        LOGGER.log(Level.INFO, "Game from {0} started.", room.getName());
+    }
+    
+    public void movePlayer(Player player, SteerAction steerAction, HandbrakeAction handbrakeAction, ThrottleAction throttleAction) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java:173]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/MovePlayerMessage.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ */
+@MessageEvent(id = MessageConstants.MOVE_PLAYER_MESSAGE_ID, handler = MovePlayerMessageHandler.class)
+public final class MovePlayerMessage extends Message {
+
+    private final String username;
+    private final SteerAction steerAction;
+    private final HandbrakeAction handbrakeAction;
+    private final ThrottleAction throttleAction;
+
+    public MovePlayerMessage(String username, SteerAction steerAction, HandbrakeAction handbrakeAction, ThrottleAction throttleAction) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/MovePlayerMessage.java:23]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerMovedMessage.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.PLAYER_MOVED_MESSAGE_ID, handler = PlayerMovedMessageHandler.class)
+public final class PlayerMovedMessage extends Message {
+
+    private final SteerAction steerAction;
+    private final HandbrakeAction handbrakeAction;
+    private final ThrottleAction throttleAction;
+
+    public PlayerMovedMessage(SteerAction steerAction, HandbrakeAction handbrakeAction, ThrottleAction throttleAction) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerMovedMessage.java:22]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 5 issues
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Review by BeeHiveJava — APPROVED** [review]
Very nice. You missed one import (as pointed out by @TehMessiah). I'll approve this, it's up to you whether you'll fix this in the future.

**Pull request merged**